### PR TITLE
feat: add dev tool to open deep links

### PR DESCRIPTION
**Pull request by johnxnguyen**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Add a developer tool to paste and open deep links directly. This makes it easier to open deep links when you have multiple Wire apps installed, you no longer have to modify the link scheme to match the bundle id.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by netbe — COMMENTED** [review]
@johnxnguyen looks good, the only thing, I wonder is `URLAction(url: url)` I see it checks either "wire" or the bundleID as scheme, should it not use WIRE_URL_SCHEME instead?

File: wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DeepLinks/DeepLinksViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            let url = URL(string: urlString),
+            (try? URLAction(url: url)) != nil
+        else {
+            error = .invalidLink
+            isShowingAlert = true
+            return
+        }
+
+        onDismiss?()
+        _ = AppDelegate.shared.appRootRouter?.openDeepLinkURL(url)
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DeepLinks/DeepLinksViewModel.swift]: Optional: Could we also inject the `appRootRouter` into this view?

File: wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DeveloperToolsViewModel.swift [hunk 2]
```diff
@@ -122,6 +122,9 @@ final class DeveloperToolsViewModel: ObservableObject {
                 })),
                 .destination(DestinationItem(title: "Configure feature flags", makeView: {
                     AnyView(DeveloperFlagsView(viewModel: DeveloperFlagsViewModel()))
+                })),
+                .destination(DestinationItem(title: "Deep links", makeView: {
+                    AnyView(DeepLinksView(viewModel: DeepLinksViewModel(onDismiss: self.onDismiss)))
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DeveloperToolsViewModel.swift]: This captures `self` (the DeveloperToolsViewModel instance), right?
Is it fine in this case?

**Review by caldrian — APPROVED** [review]

**Comment by johnxnguyen** [comment]
> @johnxnguyen looks good, the only thing, I wonder is URLAction(url: url) I see it checks either "wire" or the bundleID as scheme, should it not use WIRE_URL_SCHEME instead?

@netbe hmm yes I guess it should... but I'm not sure how to read the url scheme value. It's part of the info plist, but not under a user defined row (like we do for other build time flags). I'm wondering actually if we need to check the scheme at all?

File: wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DeveloperToolsViewModel.swift (hunk 2 above)
Comment (johnxnguyen) [reply wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DeveloperToolsViewModel.swift]: correct, I fixed it

File: wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DeepLinks/DeepLinksViewModel.swift (hunk 1 above)
Comment (johnxnguyen) [reply wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DeepLinks/DeepLinksViewModel.swift]: yes, done

**Review by netbe — APPROVED** [review]
all good and sorted, thks for the addition

**Pull request merged**